src/main/scala/nl/knaw/dans/deposit/Deposit.scala

```diff
@@ -363,66 +363,81 @@ object Deposit {
-   *                   calculated. If none are provided, SHA1 is used.
-   * @param bagInfo    the entries to be added to `bag-info.txt`
+   *                   calculated, defaults to ChecksumAlgorithm.SHA1
+   * @param bagInfo    the entries to be added to `bag-info.txt`, defaults to an empty `Map`
    * @param state      the state to be set in the `deposit.properties`' state.label
    * @param depositor  the depositor to be set in the `deposit.properties`' `depositor.userId`
-   * @param bagStore   the bagId for the `bag-dir` in this `nl.knaw.dans.bag.Deposit`
    * @return if successful, returns a `nl.knaw.dans.bag.Deposit` object representing the deposit
    *         located at `baseDir`, else returns an exception
    */
   def from(baseDir: File,
+           bagId: UUID = UUID.randomUUID(),
+           bagName: String = "bag",
            algorithms: Set[ChecksumAlgorithm] = Set(ChecksumAlgorithm.SHA1),
            bagInfo: Map[String, Seq[String]] = Map.empty,
            state: State,
-           depositor: Depositor,
-           bagStore: BagStore): Try[Deposit] = {
-    if (baseDir.exists)
-      Failure(new FileAlreadyExistsException(baseDir.toString))
+           depositor: Depositor): Try[Deposit] = {
+    lazy val depositDir = baseDir / bagId.toString
+
+    if (baseDir.notExists)
+      Failure(new NoSuchFileException(baseDir.toString))
+    else if (depositDir.exists)
+      Failure(new FileAlreadyExistsException(depositDir.toString(), null, "deposit needs to be created in this directory, but it already exists"))
     else
       for {
-        bag <- DansBag.empty(baseDir / bagStore.bagId.toString, algorithms, bagInfo)
-        properties = DepositProperties.from(state, depositor, bagStore)
-        _ <- properties.save(depositProperties(baseDir))
-      } yield new Deposit(baseDir, bag, properties)
+        bag <- DansBag.empty(depositDir / bagName, algorithms, bagInfo)
+        properties = DepositProperties.from(state, depositor, BagStore(bagId))
+        _ <- properties.save(depositProperties(depositDir))
+      } yield new Deposit(depositDir, bag, properties)
   }
 
   /**
    * Creates a new Deposit, as a parent-directory to the `payloadDir`. A new `DansBag` will be
    * created in the `Deposit`, with the bag-it files, and the data files in the `data/` directory.
    * However, no `metadata/` files will be created. These have to be added separately.
    *
+   * The new deposit will be created in `[payloadDir] / [bagId]`, with a bag in it, with the same
+   * name as `payloadDir`.
+   *
    * @param payloadDir the directory containing the payload (data) files for the bag. The `Deposit`
    *                   will be created here, and the payload files will be moved to the `data/`
    *                   directory in the new `DansBag`
+   * @param bagId      the bagId for the `bag-dir` in this `nl.knaw.dans.bag.Deposit`, defaults to
```

Comment:
it's no longer possible to specify the bag-store?

Reply:
what do you mean? See here for the list of values in `deposit.properties`: https://github.com/DANS-KNAW/easy-specs/blob/master/deposit-directory/deposit-directory.md#depositproperties

This does not contain the name of the bag-store.

Probably you're confused with 

```scala
case class BagStore(bagId: UUID, archived: Option[Boolean] = None)
```

Instead of passing the holder as an argument, you only provide the `bagId` to the methods and they make the `BagStore` object for you. If you want to set the `archived`, you can do so using `deposit.withIsArchived(true).[other-calls].save()`.